botorch/acquisition/joint_entropy_search.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+Acquisition function for joint entropy search (JES). The code utilizes the
+implementation designed for the multi-objective batch setting.
+
+"""
+
+from __future__ import annotations
+
+from typing import Any, Optional
+
+from botorch.acquisition.multi_objective.joint_entropy_search import (
+    qLowerBoundMultiObjectiveJointEntropySearch,
+)
+from botorch.acquisition.multi_objective.utils import compute_sample_box_decomposition
+from botorch.models.model import Model
+from botorch.utils.transforms import concatenate_pending_points, t_batch_mode_transform
+from torch import Tensor
+
+
+class qLowerBoundJointEntropySearch(qLowerBoundMultiObjectiveJointEntropySearch):
+    r"""The acquisition function for the Joint Entropy Search, where the batches
+    `q > 1` are supported through the lower bound formulation.
+
+    This acquisition function computes the mutual information between the observation
+    at a candidate point `X` and the optimal input-output pair.
+
+    See [Tu2022]_ for a discussion on the estimation procedure.
+    """
+
+    def __init__(
+        self,
+        model: Model,
+        optimal_inputs: Tensor,
+        optimal_outputs: Tensor,
+        maximize: bool = True,
+        hypercell_bounds: Tensor = None,
+        X_pending: Optional[Tensor] = None,
+        estimation_type: str = "LB",
+        num_samples: int = 64,
+        **kwargs: Any,
+    ) -> None:
+        r"""Joint entropy search acquisition function.
+
+        Args:
+            model: A fitted single-outcome model.
+            optimal_inputs: A `num_samples x d`-dim tensor containing the sampled
+                optimal inputs of dimension `d`. We assume for simplicity that each
+                sample only contains one optimal set of inputs.
+            optimal_outputs: A `num_samples x 1`-dim Tensor containing the optimal
+                set of objectives of dimension `1`.
+            maximize: If true, we consider a maximization problem.
+            hypercell_bounds:  A `num_samples x 2 x J x 1`-dim Tensor containing the
```

Comment:
Hmm these calling these hypercell bounds is odd, since they are not hyperrectangles, but rather lines... I guess it makes more sense in the constrained case